docs/dev/morphology/morphology.rst

```diff
@@ -0,0 +1,131 @@
+
+.. image:: Import3D-hoc.png
+
+The ``instantiate`` method of Import3D stack will create a list of HOC commands that will be executed to load the morphology.
+
+Given a simple single point soma simple Neurolucida morphology ::
+
+    ; This is the same morphology as simple.swc
+    ;
+    ;           (0, 5)
+    ; (-5, 5)----- ------ (6, 5)
+    ;             |
+    ;             |
+    ;             |
+    ;             | Type = 3
+    ;             |
+    ;             o origin
+    ;             |
+    ;             | Type = 2
+    ;             |
+    ;             |
+    ;(-5, -4)----- ------ (6, -4)
+    ;           (0, -4)
+
+
+    ("CellBody"
+    (Color Red)
+    (CellBody)
+    (0 0 0 2)
+    )
+
+    ((Dendrite)
+    (0 0 0 2)
+    (0 5 0 2)
+    (
+    (-5 5 0 3)
+    |
+    (6 5 0 3)
+    )
+    )
+
+
+    ((Axon)
+    (0 0 0 2)
+    (0 -4 0 2)
+    (
+    (6 -4 0 4)
+    |
+    (-5 -4 0 4)
+    )
+    )
+
+``instantiate`` will parse the file and create the following HOC commands
+that will be executed in the HOC interpreter, thus loading the morphology: ::
+
+    create soma[1]
+    forsec "soma" somatic.append
+    create axon[3]
+    forsec "axon" axonal.append
+    create dend[3]
+    forsec "dend" basal.append
+    forall all.append
+    soma { pt3dadd(1, 1.7484555e-07, 0, 1) }
+    soma { pt3dadd(0.94581723, -0.32469952, 0, 1) }
+    soma { pt3dadd(0.78914064, -0.61421257, 0, 1) }
+    soma { pt3dadd(0.54694813, -0.83716649, 0, 1) }
+    soma { pt3dadd(0.24548566, -0.96940023, 0, 1) }
+    soma { pt3dadd(-0.082579389, -0.99658448, 0, 1) }
+    soma { pt3dadd(-0.40169525, -0.91577339, 0, 1) }
+    soma { pt3dadd(-0.67728162, -0.73572391, 0, 1) }
+    soma { pt3dadd(-0.87947375, -0.47594735, 0, 1) }
+    soma { pt3dadd(-0.98636132, -0.16459456, 0, 1) }
+    soma { pt3dadd(-0.98636132, 0.16459462, 0, 1) }
+    soma { pt3dadd(-0.87947375, 0.47594741, 0, 1) }
+    soma { pt3dadd(-0.67728156, 0.73572391, 0, 1) }
+    soma { pt3dadd(-0.4016954, 0.91577333, 0, 1) }
+    soma { pt3dadd(-0.08257933, 0.99658448, 0, 1) }
+    soma { pt3dadd(0.2454855, 0.96940029, 0, 1) }
+    soma { pt3dadd(0.54694819, 0.83716649, 0, 1) }
+    soma { pt3dadd(0.78914052, 0.61421269, 0, 1) }
+    soma { pt3dadd(0.94581723, 0.32469946, 0, 1) }
+    soma { pt3dadd(1, 0, 0, 1) }
+    soma connect axon[0](0), 0.5
+    axon[0] { pt3dadd(0, 0, 0, 2) }
+    axon[0] { pt3dadd(0, -4, 0, 2) }
+    axon[0] connect axon[1](0), 1
+    axon[1] { pt3dadd(0, -4, 0, 2) }
+    axon[1] { pt3dadd(6, -4, 0, 4) }
+    axon[0] connect axon[2](0), 1
+    axon[2] { pt3dadd(0, -4, 0, 2) }
+    axon[2] { pt3dadd(-5, -4, 0, 4) }
+    soma connect dend[0](0), 0.5
+    dend[0] { pt3dadd(0, 0, 0, 2) }
+    dend[0] { pt3dadd(0, 5, 0, 2) }
+    dend[0] connect dend[1](0), 1
+    dend[1] { pt3dadd(0, 5, 0, 2) }
+    dend[1] { pt3dadd(-5, 5, 0, 3) }
+    dend[0] connect dend[2](0), 1
+    dend[2] { pt3dadd(0, 5, 0, 2) }
+    dend[2] { pt3dadd(6, 5, 0, 3) 
+
+
```

Comment:
Importantly, when there is a soma outline, `import3d_gui.hoc` also calls `neuron._declare_contour` to store the soma outline so that it can be used later for e.g. 3D voxelization of the soma.

Reply:
I see. Maybe this part can go into some documentation related to Import3D GUI or CellBuilder